scala-thoughts.md

```diff
@@ -0,0 +1,9 @@
+Well, that may have been the most frustrating coding experience I've had since I learned to code. Wrestling my mind into a functional programming mode (which I haven't done in 3 semesters) was challenging in itself, but the most difficult thing about this homework was dealing with SyntaxErrors and TypeErrors. Of the ~12 hours I spent on the assignment I feel like I spent maybe 3 hours designing the actual algorithms to solve the problems. While I guess it's nice that ScalaIDE tells you about TypeErrors before compile time or even runtime, their error messages are truly obfuscating and don't point you anywhere near the right direction (which is usually that I forgot to define the return type of a function (so it defaulted to unit) or that I forgot an equals sign when defining a function).
+
+To be fair, Scala does have some really nice things, things that I could appreciate even as I struggled through trying to make my code compile. The functional programming paradigm lets you use some nifty functions like map and reduce, which turn tasks that would take a long time in other languages into short work. On the other hand, I can recall a number of times when I wanted to solve a challenge using a quick for loop, but didn't have the tools to write one. Another nice feature of Scala is the sheer number of methods available in many of the utility classes - a single ScalaDoc often reads longer than 2 to 3 JavaDocs. 
+
```

Comment:
Writing a vanilla for loop is frowned upon (think (int i = 0; i < n; ++i)) and I actually found that Scala made it pretty hard to do. Even most online docs are like "Did you mean foreach or map?"